internal/branches/interactive.go

```diff
@@ -11,28 +11,31 @@ import (
 )
 
 // GetBranchName asks the user for a branch name in an interactive way
-func (b BranchProvider) GetBranchName(issueTracker domain.IssueTracker, issueIdentifier string, repo domain.Repository) (branchName string, err error) {
-	issue, err := issueTracker.GetIssue(issueIdentifier)
-	if err != nil {
-		return "", err
-	}
-
-	issueType := issueTracker.GetIssueType(issue)
+func (b BranchProvider) GetBranchName(issue domain.Issue, repo domain.Repository) (branchName string, err error) {
+	// issue, err := issueTracker.GetIssue(issueIdentifier)
+	// if err != nil {
+	// 	return "", err
+	// }
+
+	// issueType := issueTracker.GetIssueType(issue)
+	issueType := issue.Type()
 	branchType := issueType.String()
 
-	issueID := issueTracker.FormatIssueId(issue.ID)
+	// formattedID := issueTracker.FormatIssueId(issue.ID)
+	formattedID := issue.FormatID()
 
-	issueSlug := parseIssueContext(issue.Title)
+	issueSlug := parseIssueContext(issue.Title())
```

Comment:
suggestion: parseIssueContext is a bit of a strange name for its functionality. It looks like is normalizing branch name.